src/util.js

```diff
@@ -175,3 +175,39 @@ export function compare_versions(a, b) {
 
     return a_ints.length - b_ints.length;
 }
+
+/*
+ * Gets a dictionary of remote hosts with hosts as keys and addresses as values.
+ */
+export function getRemoteHosts() {
```

Comment:
this one is a bit meh. Don't take it wrongly, today I discussed this with @mvollmer and we realized there is no good way of doing this right now. We would love to have support for this in the future though. While this seems fine( I guess, haven't tested it and did not get in depth really) we have `pkg/lib/machines.js` in cockpit directly that we use in host switcher and dashboard to get list of hosts. It wraps this storage anyway. So maybe that would be a bit better approach to use that as when we change this lib we can easily sync it here as well rather than just try to rewrite this to work with whatever changes we do in Cockpit. We have some ideas and plans to change how we store and treat remote hosts, so there is a reason to be a bit worrier about it - especially since we don't call this "stable API".
(we copy things from pkg/lib here into lib/ so nothing new.If it would be impractical we can go with this for sure).

Reply:
Just to make this clear: would that mean copying the entire `pkg/lib/machines.js` file into the cockpit-podman source, as with the cockpit-components-* files? That seems a little hacky to me, it has the advantage of being able to update it more easily when it changes in cockpit, though.

Reply:
yes, thats exactly what I meant.